src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -9597,8 +9708,12 @@ private TypeWithState InferResultNullabilityOfBinaryLogicalOperator(BoundExpress
             {
                 CheckPossibleNullReceiver(receiverOpt, receiverType, checkNullableValueType: false);
             }
+            else
+            {
+                receiverOpt = null;
```

Comment:
>receiverOpt = null; [](http://example.com/codeflow?start=16&length=19)

It is not clear to me why we start pretending that there is no receiver. Perhaps, this allows us to get what we want (BTW, what is it) from ```VisitArgumentsEvaluate``` call below today. But we might get in trouble if we start using this value for something else.